core/src/coproduct.rs

```diff
@@ -581,6 +581,30 @@ impl<Head, Tail> Coproduct<Head, Tail> {
     }
 }
 
+impl<T> Coproduct<T, CNil> {
+    /// Extract the value from a coproduct with only one variant.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # fn main() {
+    /// use frunk_core::Coprod;
+    ///
+    /// type I32Only = Coprod!(i32);
+    /// let co = I32Only::inject(5);
+    ///
+    /// assert_eq!(co.extract(), 5);
+    /// # }
+    /// ```
+    #[inline(always)]
+    pub fn extract(self) -> T {
+        match self {
+            Coproduct::Inl(v) => v,
+            Coproduct::Inr(never) => match never {},
```

Comment:
Not really sure which is better, but I've always reached for [`unreachable!`](https://doc.rust-lang.org/std/macro.unreachable.html) for this sort of thing?

```suggestion
            Coproduct::Inr(never) => unreachable!(),
```

Reply:
Hmm, personally I disagree.  `match never {}` *proves* that something is unreachable, whereas it is possible for `unreachable!()` to be used in reachable code.

Reply:
Thanks both; TIL! Makes total sense to me.